### DISCUSSION TITLE
Error while using sys_probs

### DISCUSSION THREAD

**Discussion by krishnapitike** [deepmodeling/deepmd-kit#2020] [General]
Hello, I see the following error when I use sys_probs to manually set the probabilities of the data. The sum of all probabilities is equal to 1. I see the following error. The input file is provided at the end. Thanks in advance for your help.

error1:
  File "~/anaconda3/envs/deepmd/lib/python3.9/site-packages/deepmd/utils/data_system.py", line 480, in _process_sys_probs
    assert assigned_sum_prob <= 1, "the sum of assigned probability should be less than 1"
AssertionError: the sum of assigned probability should be less than 1

However if I set the total sum of probabilities to slightly smaller than 1, then I see the following error:
error2:
  File "~/anaconda3/envs/deepmd/lib/python3.9/site-packages/deepmd/utils/data_system.py", line 488,
 in _process_sys_probs
    assert np.sum(ret_prob) == 1, "sum of probs should be 1"
AssertionError: sum of probs should be 1

input file:
{
    "_comment": 		" model parameters",
    "model": {
	"type_map":		["Fe"],
	"descriptor" :{
	    "type":		"se_e2_a",
	    "sel":		[128],
	    "rcut_smth":	0.50,
	    "rcut":		6.40,
	    "neuron":		[25, 50, 100, 200],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		1,
	    "_comment":		" end descriptor"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		2,
	    "_comment":		" end fitting_net"
	},
	"_comment":		" end model"
    },

    "learning_rate" :{
	"type":			"exp",
	"decay_steps":		5000,
	"start_lr":		0.001,
	"stop_lr":		3.51e-8,
	"_comment":		" end learning_rate"
    },

    "loss" :{
	"type":			"ener",
	"start_pref_e":		0.02,
	"limit_pref_e":		1,
	"start_pref_f":		1000,
	"limit_pref_f":		1,
	"start_pref_v":		0.2,
	"limit_pref_v":		1,
	"_comment":		" end loss"
    },

    "training" : {
	"training_data": {
	    "systems":		[
                                 "data_npy/Fe19_tr/",
                                 "data_npy/Fe37_tr/",
                                 "data_npy/Fe30_tr/",
                                 "data_npy/Fe34_tr/",
                                 "data_npy/Fe40_tr/",
                                 "data_npy/Fe129_tr/",
                                 "data_npy/Fe16_tr/",
                                 "data_npy/Fe32_tr/",
                                 "data_npy/Fe45_tr/",
                                 "data_npy/Fe54_tr/",
                                 "data_npy/Fe25_tr/",
                                 "data_npy/Fe15_tr/",
                                 "data_npy/Fe126_tr/",
                                 "data_npy/Fe20_tr/",
                                 "data_npy/Fe127_tr/",
                                 "data_npy/Fe13_tr/",
                                 "data_npy/Fe42_tr/",
                                 "data_npy/Fe41_tr/",
                                 "data_npy/Fe17_tr/",
                                 "data_npy/Fe36_tr/",
                                 "data_npy/Fe22_tr/",
                                 "data_npy/Fe21_tr/",
                                 "data_npy/Fe43_tr/",
                                 "data_npy/Fe33_tr/",
                                 "data_npy/Fe27_tr/",
                                 "data_npy/Fe26_tr/",
                                 "data_npy/Fe18_tr/",
                                 "data_npy/Fe128_tr/",
                                 "data_npy/Fe23_tr/",
                                 "data_npy/Fe31_tr/",
                                 "data_npy/Fe2_tr/",
                                 "data_npy/Fe29_tr/",
                                 "data_npy/Fe38_tr/",
                                 "data_npy/Fe39_tr/",
                                 "data_npy/Fe12_tr/",
                                 "data_npy/Fe24_tr/",
                                 "data_npy/Fe44_tr/",
                                 "data_npy/Fe28_tr/",
                                 "data_npy/Fe11_tr/",
                                 "data_npy/Fe14_tr/",
                                 "data_npy/Fe35_tr/",
                                 "data_npy/Fe130_tr/"
                                ],
            "sys_probs":        [
                                 0.01 ,
                                 0.02 ,
                                 0.01 ,
                                 0.01 ,
                                 0.02 ,
                                 0.1 ,
                                 0.01 ,
                                 0.01 ,
                                 0.02 ,
                                 0.2 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.1 ,
                                 0.01 ,
                                 0.02 ,
                                 0.02 ,
                                 0.01 ,
                                 0.02 ,
                                 0.01 ,
                                 0.01 ,
                                 0.02 ,
                                 0.02 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.02 ,
                                 0.02 ,
                                 0.01 ,
                                 0.01 ,
                                 0.02 ,
                                 0.01 ,
                                 0.01 ,
                                 0.01 ,
                                 0.02 ,
                                 0.1
                                ],
            "batch_size":       1,
	    "_comment":		" end training_data"
	},
	"validation_data":{
	    "systems":		[
                                 "data_npy/Fe19_te/",
                                 "data_npy/Fe37_te/",
                                 "data_npy/Fe30_te/",
                                 "data_npy/Fe34_te/",
                                 "data_npy/Fe40_te/",
                                 "data_npy/Fe129_te/",
                                 "data_npy/Fe16_te/",
                                 "data_npy/Fe32_te/",
                                 "data_npy/Fe45_te/",
                                 "data_npy/Fe54_te/",
                                 "data_npy/Fe25_te/",
                                 "data_npy/Fe15_te/",
                                 "data_npy/Fe126_te/",
                                 "data_npy/Fe20_te/",
                                 "data_npy/Fe127_te/",
                                 "data_npy/Fe13_te/",
                                 "data_npy/Fe42_te/",
                                 "data_npy/Fe41_te/",
                                 "data_npy/Fe17_te/",
                                 "data_npy/Fe36_te/",
                                 "data_npy/Fe22_te/",
                                 "data_npy/Fe21_te/",
                                 "data_npy/Fe43_te/",
                                 "data_npy/Fe33_te/",
                                 "data_npy/Fe27_te/",
                                 "data_npy/Fe26_te/",
                                 "data_npy/Fe18_te/",
                                 "data_npy/Fe128_te/",
                                 "data_npy/Fe23_te/",
                                 "data_npy/Fe31_te/",
                                 "data_npy/Fe2_te/",
                                 "data_npy/Fe29_te/",
                                 "data_npy/Fe38_te/",
                                 "data_npy/Fe39_te/",
                                 "data_npy/Fe12_te/",
                                 "data_npy/Fe24_te/",
                                 "data_npy/Fe44_te/",
                                 "data_npy/Fe28_te/",
                                 "data_npy/Fe11_te/",
                                 "data_npy/Fe14_te/",
                                 "data_npy/Fe35_te/",
                                 "data_npy/Fe130_te/"
                                ],
	    "batch_size":	1,
	    "numb_btch":	1,
	    "_comment":		" end validation_data"
	},
	"numb_steps":		2000000,
	"seed":			3,
	"disp_file":		"lcurve.out",
	"disp_freq":		100,
	"save_freq":		1000,
	"_comment":		" end training"
    },

    "_comment":			" end input.json"
}

**Comment by njzjz** [comment]
This has been fixed in #1919.
